website/src/content/docs/docs/getting-started/code-generation-csharp.md

```diff
@@ -0,0 +1,312 @@
+3. The `MockRegistration.cs` file registers your implementations with the dependency injection container
+4. When a request comes in, the controller receives it and calls your implementation
+
+If you need to register additional services or dependencies, you would add them to the `MockRegistration.cs` file.
+
+## 8. Adding your business logic
+
+1. Locate the implementation file for your service (e.g., `Widgets.cs`)
+
+2. Update the implementation methods with your actual business logic. For example:
+
+```csharp
+public async Task<Widget[]> ListAsync()
+{
+    // Replace the mock implementation with your actual database query
+    return new Widget[]
+    {
+        new Widget { Id = "1", Weight = 10, Color = "red" },
+        new Widget { Id = "2", Weight = 15, Color = "blue" }
+    };
+}
+```
+
+3. The implementation file won't be overwritten when you recompile your TypeSpec, preserving your business logic.
+
+## 9. Evolving your API
+
+As your API evolves, you'll need to update your TypeSpec definition and regenerate code:
+
+1. Modify your TypeSpec definition (e.g., add new models or operations)
+
+2. Recompile to update the generated code:
+
+```bash
+tsp compile
+```
+
+This updates the generated files (controllers, interfaces, models) but preserves your implementation files.
+
+3. If you've added entirely new resources that require new implementation files:
+
+```bash
+npx hscs scaffold main.tsp
+```
+
+This is particularly useful when you've added new interfaces in your TypeSpec. For example, if you add a new `Categories` interface:
+
+```typespec
+@route("/categories")
+@tag("Categories")
+interface Categories {
+  /** List categories */
+  @get list(): CategoryList | Error;
+  // More operations...
+}
+```
+
+When you run the scaffolding step again:
+
+- It will create new files for the new `Categories` interface (`CategoriesController.cs`, `ICategories.cs`, and `CategoriesImpl.cs`)
+- It won't overwrite your existing `Widgets.cs` with your custom business logic
+- This allows you to incrementally add new resources without losing your existing implementations
+
+## 10. Advanced customization options
+
+When scaffolding your service, you can use additional options for customization:
+
+```bash
+npx hscs scaffold main.tsp --help
+```
+
+Some useful options include:
+
+- `--project-name <name>`: Set a custom project name
+- `--https-port <port>`: Set a different HTTPS port
+- `--output <path>`: Generate files to a different location
+- `--overwrite`: Overwrite existing files (use with caution)
+  - Normally, implementation files that already exist won't be overwritten
+  - Use this flag if you want to reset your implementations back to the generated mock versions
+
+## Next steps
+
+- Explore the included README and documentation in the `docs` folder of your generated project
+- Add authentication mechanisms to your service
+- Implement data validation and error handling
+- Connect your implementation to a database
```

Comment:
I think we need an additional section in this document (or an additional document) that allows you to create a service when you are using the standalone tsp executable (without node.js / npm separately installed.

In this case, you can use:

```bash
// for the first scaffolding
> tsp compile . --option @typespec/http-server-csharp.emit-mocks=mocks-and-project-files --option @typespec/http-server-csharp.use-swaggerui=true --option @typespec/http-server-csharp.overwrite=true  

// for incremental scaffolding
> tsp compile . --option @typespec/http-server-csharp.emit-mocks=mocks-and-project-files
```

Let me know if you prefer I add this doc

Reply:
Given that we're putting out the standalone as an experimental feature, I'd prefer documentation for it to live elsewhere to avoid confusion. If/when we make it official then we can add that content to this doc.